### PR TITLE
Task/dwr 1853 target appsettings

### PR DESCRIPTION
**Pull request by jtreuting**
Bring server side config settings for target report to the client

### CODE REVIEW

File: webapp/src/main/resources/application.yml [hunk 1]
```diff
@@ -129,3 +129,7 @@ reporting:
   # english-learners should be lep or elas
   english-learners:
     - elas
+
+  target-report:
```
Comment (jtreuting) [inline webapp/src/main/resources/application.yml]: this is a duplicate of what @agorina did in the aggregate-service.  Should it be moved to somewhere common?  thoughts?

File: webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.html [hunk 2]
```diff
@@ -139,23 +139,23 @@
             <!-- pick the label color -->
             <span [ngSwitch]="getTargetReportingLevelString(rowData.standardMetRelativeLevel)">
               <span *ngSwitchCase="'Above'"
-                    class="label sb-iab-green">
+                    class="label label-lg sb-iab-green">
```
Comment (jtreuting) [inline webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.html]: not related, but bigger labels on the report output before i forgot

File: webapp/src/main/webapp/src/app/app-settings.service.ts [hunk 3]
```diff
@@ -28,6 +28,10 @@ export class ApplicationSettingsService {
           code: serverSettings.state.code,
           name: serverSettings.state.name
         },
+        targetReport: {
+          insufficientDataCutoff: serverSettings.targetReport.insufficientDataCutoff,
+          minNumberOfStudents: serverSettings.targetReport.minNumberOfStudents
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/app-settings.service.ts]: would prefer `minimumStudentCount`

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts [hunk 4]
```diff
@@ -11,10 +11,20 @@ import { Subgroup } from '../../aggregate-report/subgroup/subgroup';
...
+
   constructor(private examStatisticsCalculator: ExamStatisticsCalculator,
               private subgroupMapper: SubgroupMapper) {
   }
 
+  /**
+   * Set the standard error cutoff used when determining if the data is sufficient or not
+   * @param {number} cutoff
+   */
+  setInsufficientDataCutoff(cutoff: number): void {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: nit: you could use a setter:
```typescript
set insufficientDataCutoff(cutoff: number): void {
  //...
}
```

Comment (jtreuting) [reply webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: i always forget about those in typescript, thanks

File: webapp/src/main/webapp/src/app/app-settings.service.ts (hunk 3 above)
Comment (jtreuting) [reply webapp/src/main/webapp/src/app/app-settings.service.ts]: that's fine with me, was being consistent with the existing setting, but i'll change it on the client side

File: webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.ts [hunk 5]
```diff
@@ -128,12 +125,15 @@ export class TargetReportComponent implements OnInit {
...
-
-    ).subscribe(([ allTargets, targetScoreExams, subgroupOptions ]) => {
+      this.applicationSettingsService.getSettings()
+    ).subscribe(([ allTargets, targetScoreExams, subgroupOptions, applicationSettings ]) => {
       this.targetScoreExams = targetScoreExams;
       this.subgroupOptions = subgroupOptions;
       this.allTargets = allTargets;
 
+      this.minimumStudents = applicationSettings.targetReport.minNumberOfStudents;
+      this.targetStatisticsCalculator.setInsufficientDataCutoff(applicationSettings.targetReport.insufficientDataCutoff);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.ts]: i think ideally we pass this in as a method param, or the service makes the call to get this value at method invocation time and returns an observable

Comment (jtreuting) [reply webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.ts]: i originally was passing it in, but it is used about 4 method calls within the main one and that didn't feel right.  it's not really a concern of the first call to generate the aggregate rows.  it also feels weird to have what is essentially a set of helper methods call a service and get the observable.

File: webapp/src/main/resources/application.yml (hunk 1 above)
Comment (malaffoon) [reply webapp/src/main/resources/application.yml]: As discussed offline, copy and paste is okay for this. Please add to configuration samples in RDW.

Comment (jtreuting) [reply webapp/src/main/resources/application.yml]: yep, done: https://github.com/SmarterApp/RDW/pull/32

**Pull request merged**